orderer/common/multichannel/registrar.go

```diff
@@ -608,19 +619,41 @@ func (r *Registrar) JoinChannel(channelID string, configBlock *cb.Block, isAppCh
 	if err != nil {
 		return types.ChannelInfo{}, err
 	}
+	defer func() {
```

Comment:
@tock-ibm What about the case where the initLedgerResourcesClusterConsenter fails? That will leave around the ledger for the channel. Do you think newLedgerResources should be updated to remove the ledger if the call to ledgerFactory.GetOrCreate() fails?

Reply:
That is a good point. `GetOrCreate` is the last thing in `newLedgerResources` so I am not too worried about that. Hopefully `GetOrCreate`  does not leave an empty ledger behind if it fails. `initLedgerResourcesClusterConsenter` is used in other places as well, in which the ledger is not empty, so we cant effort to cleanup within it as well.
However if  `initLedgerResourcesClusterConsenter` fails after `newLedgerResources`, that requires cleanup in this method, that is correct. 

Here is an idea: how about putting the `defer func()`  from L622 just before `initLedgerResourcesClusterConsenter` (to L617)? 
It might try to remove a non-existing ledger, but that is not an error according to the blockstore_provider.go#L131

Reply:
Done